mlir_graphblas/engine.py

```diff
@@ -533,12 +541,139 @@ def __init__(self, cli_executable=None, cli_options=None, llvmlite_engine=None):
             backing_mod = llvm.parse_assembly("")
             llvmlite_engine = llvm.create_mcjit_compiler(backing_mod, target_machine)
         self._engine = llvmlite_engine
+        self.tmp_dir = tempfile.TemporaryDirectory()
+        self.c_compiler = distutils.ccompiler.new_compiler()
         self._cli = MlirOptCli(cli_executable, cli_options)
         self.name_to_callable: Dict[str, Callable] = {}
+        return
+
+    def profiled_function(
+        self, main_callable: Callable, symbol_to_profile: str
```

Comment:
Can we have an option to pass in the name of the profile file to record, so that it can be analyzed offline?

Reply:
https://github.com/metagraph-dev/mlir-graphblas/pull/98/commits/e39f25858cb5bd6abf266bde7eb30f7cf8c67bae makes it so that when we add the function via `MlirJitEngine.add`, we can also specify a `profile_result_directory`. The generated `.o`, `.so`, and `perf-*.data` files will be stored there.